hooks/useNewMqlQuery.ts

```diff
@@ -303,41 +306,54 @@ export default function useNewMqlQuery({
     }
 
     dispatch({ type: "postQueryStart" });
-    createMqlQuery({
-      metrics: [metricName],
-      groupBy: formState.groupBy || [],
-      where: clearEmptyConstraints(formState.where),
-      pctChange: formState.pctChange,
-      timeGranularity: formState?.granularity ? granularityToTimeGranularityMap[formState?.granularity] : null,
-      addTimeSeries: true,
-      startTime: formState.startTime,
-      endTime: formState.endTime,
-      order: formState.order,
-      limit: formState.limit
-    }).then(({ data, error }) => {
-      if (data?.createMqlQuery?.query?.status === MqlQueryStatus.Successful) {
-        dispatch({
-          type: "postQueryCachedResultsSuccess",
-          data,
-          limit,
-          handleCombinedError,
-        });
-      } else {
-        if (data?.createMqlQuery?.id) {
-          dispatch({
-            type: "postQuerySuccess",
-            queryId: data?.createMqlQuery?.id,
-          });
-        } else if (error) {
+
+    const doCreateMqlQuery = () => {
+      createMqlQuery({
+        metrics: [metricName],
+        groupBy: formState.groupBy || [],
+        where: clearEmptyConstraints(formState.where),
+        pctChange: formState.pctChange,
+        timeGranularity: formState?.granularity ? granularityToTimeGranularityMap[formState?.granularity] : null,
+        addTimeSeries: true,
+        startTime: formState.startTime,
+        endTime: formState.endTime,
+        order: formState.order,
+        limit: formState.limit
+      }).then(({ data, error }) => {
+        if (data?.createMqlQuery?.query?.status === MqlQueryStatus.Successful) {
           dispatch({
-            type: "postQueryFail",
-            errorMessage: getErrorMessage(error),
+            type: "postQueryCachedResultsSuccess",
+            data,
+            limit,
+            handleCombinedError,
           });
-          handleCombinedError(error);
+        } else {
+          if (data?.createMqlQuery?.id) {
+            dispatch({
+              type: "postQuerySuccess",
+              queryId: data?.createMqlQuery?.id,
+            });
+          } else if (error) {
+            if (retries > 0 && state.retries !== retries) {
+              setTimeout(() => {
+                dispatch({ type: "retryFetchResults" });
+                doCreateMqlQuery();
```

Comment:
do we need to decrement retries here?

Reply:
That happens in the retryFetchResults action handler.